documentation for cross platform executables

bug-or-feature opened the pull request:
This PR illustrates a simple way to create executable commands direct from existing python functions. The mechanism is provided by the [setup tools](https://python-packaging.readthedocs.io/en/latest/command-line-scripts.html#the-console-scripts-entry-point), and configured in setup.py. The benefits are:

- no need for additional code or scripts
- works for linux, macos, windows
- standard Python
- no need to manipulate PATH or other env variables
- name completion
- works with any flavour of virtualenv

I found about this stuff this from the [click](https://click.palletsprojects.com/en/8.1.x) docs. Click is a transitive dependency in the project, via Flask. They give [a few more benefits](https://click.palletsprojects.com/en/8.1.x/setuptools/#) I'm not sure I fully understand

@robcarver17 I've left this PR in draft, there's many ways to proceed, depending how you feel about it:

- the change in this PR just creates shims for the `interactive_*` scripts. We could leave it at that, and add a note in the docs so users who want it can follow the pattern for other stuff. It would mostly benefit [non linux users](https://github.com/robcarver17/pysystemtrade/discussions/576)
- If used to its fullest extent, all the executable files could have their own entry
- all the files in `sysproduction/linux/scripts` could be removed completely. Would obviously be a quite intrusive change, crontab changed, docs changed etc
- some combination of both, in parallel. may need to be careful of name clashes
- something else
- leave it, do nothing
- another potential option that I use in my own fork is to use click to create a top level script for all the interactive commands. So to run the interactive roll tool, I run `pst r`, and to run the historical price check i run `pst h`, and so on. The script is [here](https://github.com/bug-or-feature/pysystemtrade-fsb/blob/develop/sysproduction/pst.py)

robcarver17 commented:
I guess the solution I'd be happiest with in the short term is something that ended up in a directory called sysproduction/windows/scripts or sysproduction/not-linux/scripts.

bug-or-feature commented:
> I guess the solution I'd be happiest with in the short term is something that ended up in a directory called sysproduction/windows/scripts or sysproduction/not-linux/scripts

I don't think that is possible with this mechanism. I could add a section in the production docs for non linux users instead, and reference it there?

robcarver17 commented:
That's fine. Basically I'd rather not disturb the linux scripts, since as you say there is a lot of documentation references - unless the scripts remained in place but were generated differently?

bug-or-feature commented:
> unless the scripts remained in place but were generated differently?

I don't think that is possible either. 

I have reverted the setup.py change, and added a docs section instead. We can always add it later if you change your mind. I don't mind updating the docs
